lib/pat.c

```diff
@@ -819,7 +822,10 @@ _grn_pat_add(grn_ctx *ctx, grn_pat *pat, const uint8_t *key, uint32_t size, uint
           p0 = &rn0->lr[nth_bit(key, c0, len)];
         }
       } else {
-        if (!(s = pat_node_get_key(ctx, pat, rn0))) { return GRN_ID_NIL; }
+        if (!(s = pat_node_get_key(ctx, pat, rn0))) {
+          ERR(GRN_UNKNOWN_ERROR, "[pat][key] failed to get key from node");
```

Comment:
It's better that we don't use the same message in different location.
It's difficult to detect logged location.